src/nbl/video/COpenGL_LogicalDevice.h

```diff
@@ -135,7 +135,33 @@ class COpenGL_LogicalDevice : public IOpenGL_LogicalDevice
 
         return retval;
     }
+    core::smart_refctd_ptr<IGPUImage> createImage(IGPUImage::SCreationParams&& params) override final
+    {
+        if (!asset::IImage::validateCreationParameters(params))
+            return nullptr;
+        if constexpr (IsGLES)
+        {
+            if (params.type == IGPUImage::ET_1D)
+                return nullptr;
+        }
 
+        core::smart_refctd_ptr<IGPUImage> retval;
+
+        SRequestImageCreate2 reqParams;
+
+        reqParams.mreqs.size = 0u; // TODO(Erfan) some approx of image size -> considering dimensions, mipLevels, arrayLayers, samples, minImageGranularity and texelBlockInfo (see ImageUploadUtility)
```

Comment:
we should have some utility in `asset` namespace to setup the Regions array (calculate offsets into a CPUBuffer with provided alignment restrictions) for a blank ICPUImage (loaders, and other utilities would benefit).

Then we could reuse here (simply by iterating the regions array, finding the one with highest offset+size and rounding that up to whatever alignment is necessary).